src/VisualStudio/Core/Def/Implementation/ProjectSystem/VisualStudioAnalyzer.cs

```diff
@@ -135,5 +136,37 @@ public Assembly LoadFromPath(string fullPath)
                 return _analyzer._loader.LoadFromPath(fullPath);
             }
         }
+
+        /// <summary>
+        /// This custom <see cref="AnalyzerReference"/>, just wraps an existing <see cref="UnresolvedAnalyzerReference"/>,
+        /// but ensure that we start listening to the file for changes once we've actually observed it, so that if the
+        /// file then gets created on disk, we are notified.
+        /// </summary>
+        private class VisualStudioUnresolvedAnalyzerReference : AnalyzerReference
+        {
+            private readonly UnresolvedAnalyzerReference _underlying;
+            private readonly VisualStudioAnalyzer _visualStudioAnalyzer;
+
+            public VisualStudioUnresolvedAnalyzerReference(string fullPath, VisualStudioAnalyzer visualStudioAnalyzer)
+            {
+                _underlying = new UnresolvedAnalyzerReference(fullPath);
+                _visualStudioAnalyzer = visualStudioAnalyzer;
+            }
+
+            public override string FullPath
+                => _underlying.FullPath;
+
+            public override object Id
+                => _underlying.Id;
+
+            public override ImmutableArray<DiagnosticAnalyzer> GetAnalyzers(string language)
+            {
+                _visualStudioAnalyzer._tracker.EnsureSubscription();
```

Comment:
Nested types are tricky like that.  Note that this is the same pattern as `AnalyzerAssemblyLoaderThatEnsuresFileBeingWatched` lower in the file.